pinakes/main/catalog/serializers.py

```diff
@@ -429,14 +429,29 @@ class SharingRequestSerializer(serializers.Serializer):
         many=True, queryset=Group.objects.all(), help_text="List of group IDs."
     )
 
+    valid_permissions = None
+
     def validate_permissions(self, value):
-        valid_scopes = self.context.get("valid_scopes", [])
-        invalid_scopes = set(value).difference(valid_scopes)
-        if invalid_scopes:
-            raise serializers.ValidationError(
-                "Unexpected permissions: {}".format(", ".join(invalid_scopes))
-            )
-        return value
+        value_set = set(value)
+        for permissions in self.valid_permissions:
+            if value_set == permissions:
+                return value
+        raise serializers.ValidationError(
+            "Unexpected permissions: {}".format(", ".join(value))
+        )
+
+
+class SharingRequestSerializer(_SharingRequestSerializerBase):
+    valid_permissions = [
+        {"read", "order"},
+        {"read", "order", "update", "delete"},
+    ]
+
+
+class UnsharingRequestSerializer(_SharingRequestSerializerBase):
+    valid_permissions = [
+        {"read", "order", "update", "delete"},
```

Comment:
Shouldn't the `valid_permissions` lists be the same? If someone shares a portfolio with read+order permissions, and then wants to unshare later, they would have to pass in the full list instead of just the permissions that were passed in in the first place, and I'm not sure if that would work because then we'd be trying to remove permissions that weren't originally applied.

I think that's how the UI is currently calling the API endpoint, with the specific permissions that were shared in the first place. If we only allow /unshare to be called with all four permissions even if it was shared with only two, then I think we could simply change the signature of the API call to not even take the permissions as an argument (only take the group you're unsharing to). Then we would just always try to remove all four permissions by default.

Might have to wait for @lgalis to get back to confirm this.